api/context.go

```diff
@@ -378,3 +452,38 @@ func Handle404(w http.ResponseWriter, r *http.Request) {
 	l4g.Error("%v: code=404 ip=%v", r.URL.Path, GetIpAddress(r))
 	RenderWebError(err, w, r)
 }
+
+func CreateOAuthSession(token string) (*model.Session, *model.AppError) {
```

Comment:
Should we be differentiating OAuth sessions from regular sessions? This would allow users to set special permissions on these sessions. (Such as saying no you can't change my username, or post in specific channels)